### DISCUSSION TITLE
[firebase_auth] Sign in with Apple on native does not give email and full name. How to get email and fullName?

### DISCUSSION THREAD

**Discussion by adhikari-suman** [firebase/flutterfire#9835] [Q&A]
Hi, I am trying to signinwith Apple on native platforms.

```dart
import 'package:firebase_auth/firebase_auth.dart';

Future<UserCredential> signInWithApple() async {
  final appleProvider = AppleAuthProvider();
  if (kIsWeb) {
    await FirebaseAuth.instance.signInWithPopup(appleProvider);
  } else {
    await FirebaseAuth.instance.signInWithProvider(appleProvider);
  }
}
```

Source: [Firebase Federation Auth & Social Login](https://firebase.google.com/docs/auth/flutter/federated-auth)

I have added scopes to appleProvider as well as below:

```dart
import 'package:firebase_auth/firebase_auth.dart';

Future<UserCredential> signInWithApple() async {
  final appleProvider = AppleAuthProvider();

    appleProvider.addScope('email');
    appleProvider.addScope('name');

  if (kIsWeb) {
    await FirebaseAuth.instance.signInWithPopup(appleProvider);
  } else {
    await FirebaseAuth.instance.signInWithProvider(appleProvider);
  }
}

```

Even still, I am only able to fetcha the email but no display name comes at all. How can I request for email and fullname for user? It is required for us to register a new user to our backend.

**Comment by goldenmoon67** [comment]
Hi, did you solved this issue

**Reply by adhikari-suman** [reply]
Yes, I resolved it uisng the sign_in_with_apple package and set the details myself in firebase. It was a hassle not to have it but I couldn't wait for firebase to update their package.

**Reply by goldenmoon67** [reply]
Oh okay. I got it. But I have to use fireabase for my backend :(

**Reply by adhikari-suman** [reply]
I didn't mean not to use firebase as a backend. I'm only suggesting you to login to apple in flutter using the package `sign_in_with_apple`. Then, you fetch the user details deemed necessary like email and name and idToken. Then, you register/login the user in firebase using firebase_auth. Then, once you are signed in, you set the email and fullName property manully in firebase  current user. This way when you fetch the details in backend, you'll have them.

**Reply by goldenmoon67** [reply]
I get it. Thanks a lot. This is really helpfull

**Reply by adhikari-suman** [reply]
Glad I could help.

**Reply by parulgupta27** [reply]
@ExpertKiD  I am getting email and password as null by using sign_in_with_apple package, and by using firebase AppleAuthProvider i am losing contact to debug console in simulator. Any Idea?
